Optimize ReadMe and run-webkit-app usage

Pancf opened the pull request:
#### e2f7bd55eda49489399eb5bf0b10b71cbb6de02c
<pre>
Optimize ReadMe and run-webkit-app usage
<a href="https://bugs.webkit.org/show_bug.cgi?id=278906">https://bugs.webkit.org/show_bug.cgi?id=278906</a>

Reviewed by Jonathan Bedard.

Add the guide of running applications in iOS simulator to ReadMe.
Update the utility `run-webkit-app` usage output.

* ReadMe.md:
* Tools/Scripts/webkitdirs.pm:
(determineXcodeSDKPlatformName):
(printHelpAndExitForRunAndDebugWebKitAppIfNeeded):

Canonical link: <a href="https://commits.webkit.org/285197@main">https://commits.webkit.org/285197@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/19a498b597fb9b0ceec5b6d87d92f66a46e7e18b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71817 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51230 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24591 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75933 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23022 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73932 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59031 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22842 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56686 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15188 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74883 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46471 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61864 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37145 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43132 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19332 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21364 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/64942 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65040 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19694 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77652 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/71067 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16051 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18880 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/64435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16095 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61897 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64423 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15875 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12594 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6233 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/92852 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47030 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20460 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48101 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49385 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47843 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -3011,7 +3012,7 @@ Options specific to macOS:
 Options specific to iOS:
-  --iphone-simulator                Run the app in an iPhone Simulator
+  --ios-simulator                   Run the app in an iPhone Simulator
   --ipad-simulator                  Run the app in an iPad Simulator
```
Someone will need to check the actual behavior as part of reviewing, but it would be very incorrect if we opposed "ios" and "ipad" here. Other tools either have "iphone" and "ipad", or only "ios". 

Given that this tool needs to launch a simulator, differentiating between these makes sense, so I think that the usage text describes how it should probably behave in reality, even if it does not.

JonWBedard commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -3011,7 +3012,7 @@ Options specific to macOS:
 
 Options specific to iOS:
-  --iphone-simulator                Run the app in an iPhone Simulator
+  --ios-simulator                   Run the app in an iPhone Simulator
```
To add on to @aproskuryakov's comment here, this is case where the documentation reflects what we _should_ be doing, but the code in the script is actually wrong. Looking at `Tools/Scripts/webkitdirs.pm`, the good news is that a naive find-and-replace of `--ios-simulator` -> `--iphone-simulator` in `Tools/Scripts/webkitdirs.pm` will get the desired behavior, and that's what we should do here.

JonWBedard replied:
Sorry, to add to this, @aproskuryakov and I were looking at the code. This is going to require adding `--iphone-simulator` on line `987`, but keeping `ios-simulator` because this code is relied on by many other tools which use `--ios-simulator` (builds, in particular, which use that argument correctly as our iPhone and iPad builds are the same)

Pancf replied:
ok, I will revert here and update the webkitdirs.pm instead. thank you for your inspecting:)

Pancf replied:
@JonWBedard already updated, ptal

JonWBedard commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -984,9 +984,9 @@ sub determineXcodeSDKPlatformName {
         $simulatorIdiom = 'iPhone';
     }
-    if (checkForArgumentAndRemoveFromARGV("--ios-simulator")) {
+    if (checkForArgumentAndRemoveFromARGV("--iphone-simulator")) {
```
Looks like this needs to be:
```
if (checkForArgumentAndRemoveFromARGV("--iphone-simulator") || checkForArgumentAndRemoveFromARGV("--ios-simulator"))
```
at least for a little while, since we can't easily synchronize a builedbot restart with this change.

Pancf committed:
Optimize ReadMe and run-webkit-app usage
https://bugs.webkit.org/show_bug.cgi?id=278906

Reviewed by Jonathan Bedard.

Add the guide of running applications in iOS simulator to ReadMe.
Update the utility `run-webkit-app` usage output.

* ReadMe.md:
* Tools/Scripts/webkitdirs.pm:
(determineXcodeSDKPlatformName):
(printHelpAndExitForRunAndDebugWebKitAppIfNeeded):

Canonical link: https://commits.webkit.org/285197@main
